Core/include/Acts/Navigation/DetectorNavigator.hpp

```diff
@@ -405,13 +406,12 @@ class DetectorNavigator {
 
     // Sort properly the surface candidates
     auto& nCandidates = nState.surfaceCandidates;
-    std::sort(nCandidates.begin(), nCandidates.end(),
-              [&](const auto& a, const auto& b) {
-                // The two path lengths
-                ActsScalar pathToA = a.objectIntersection.pathLength();
-                ActsScalar pathToB = b.objectIntersection.pathLength();
-                return pathToA < pathToB;
-              });
+    std::ranges::sort(nCandidates, [&](const auto& a, const auto& b) {
```

Comment:
We could also consider using the projector idiom more with `std::ranges`:
```c++
std::ranges::sort(nCandidates, {}, [](const auto &c) { return a.objectIntersection.pathLength(); });
```
Would be even shorter I think.

Reply:
That looks really nice. I used `std::tie` for the cases, where we had a chain of multiple dimensions to compare.

Reply:
* `std::make_tuple`, weil `tie` nur eine lhs reference erzeugt, die man nicht returnen kann oder sowas.

Reply:
Hmm I don't get it fully, what `std::tie` has to do with this actually...

Reply:
Not directly with that one, but it was an option, to use a lexicographic order inside a projection.